passes/opt/opt_merge.cc

```diff
@@ -312,42 +273,51 @@ struct OptMergeWorker
 			}
 
 			did_something = false;
-			std::map<RTLIL::Cell*, RTLIL::Cell*, CompareCells> sharemap(CompareCells(this));
+			std::unordered_map<std::string, RTLIL::Cell*> sharemap;
```

Comment:
Why not use `dict`? Otherwise LGTM.

Reply:
Come to think of it, is there any reason not to hash directly to a `size_t` instead of a SHA-1?